spec/osut_tests_spec.rb

```diff
@@ -1680,8 +1686,71 @@
 
     expect(wall3.windowToWallRatio).to be_within(TOL).of(wwr * 2)
 
+    # Fetch transform if wall4 vertices were to 'align'.
+    tr = OpenStudio::Transformation.alignFace(wall4.vertices)
+    a_wall4 = tr.inverse * wall4.vertices
+    ymax = a_wall4.map(&:y).max
+    xmax = a_wall4.map(&:x).max
+    xmid = xmax / 2 # centreline
+
+    # Add 4x sub surfaces (with frame & dividers) to wall4:
+    #   1. w1: 0.8m-wide opening (head defaulted to HEAD, sill @0m)
+    #   2. w2: 0.4m-wide sidelite, to the immediate right of w2 (HEAD, sill@0)
+    #   3. t1: 0.8m-wide transom above w1 (0.4m in height)
+    #   4. t2: 0.5m-wide transom above w2 (0.4m in height)
+    #
+    # All 4x sub surfaces are intended to share frame edges (once frame &
+    # divider frame widths are taken into account). Postulating a 50mm frame,
+    # meaning 100mm between w1, w2, t1 vs t2 vertices. In addition, all 4x
+    # openings (grouped together) should align towards the left of wall4,
+    # leaving a 200mm gap between the left vertical wall edge and the left
+    # frame jamb edge of w1 & t1. First initialize Frame & Divider object.
```

Comment:
Tests/demos:
- head/sill alignment of new sub surface with those of an existing one (e.g. bottom wwr% strip)
- recovery of OSut default head/sill (if missing), e.g. top wwr% strip
- left/right offset of 4x grouped sub surfaces (from _base_ surface centreline)
- frame & divider enabled (4x grouped sub surfaces)
- E+ simulations (no related warnings or errors)

<img width="1346" alt="seb_offset" src="https://user-images.githubusercontent.com/98104985/236543794-63168641-6e45-4501-9efc-bbc4095945cf.png">

Reply:
This looks pretty cool!